sources/ClangSharp.PInvokeGenerator/PInvokeGenerator.VisitPreprocessedEntity.cs

```diff
@@ -34,7 +34,7 @@ private unsafe void VisitMacroDefinitionRecord(MacroDefinitionRecord macroDefini
             var macroName = $"ClangSharpMacro_{macroDefinitionRecord.Spelling}";
 
             _ = _fileContentsBuilder.Append('\n');
-            _ = _fileContentsBuilder.Append($"const auto {macroName} = ");
+            _ = _fileContentsBuilder.Append($"const __auto_type {macroName} = ");
```

Comment:
This overall LGTM, but there should be an explicit test added to https://github.com/dotnet/ClangSharp/blob/main/tests/ClangSharp.PInvokeGenerator.UnitTests/CTest.cs ensuring macro handling "works".

You can probably just copy the `MacroTest` from: https://github.com/dotnet/ClangSharp/blob/main/tests/ClangSharp.PInvokeGenerator.UnitTests/CSharpLatestWindows/VarDeclarationTest.cs#L72-L91 and ensure the `C` specific command line args are passed in.

It would be likewise be nice if we could continue using `auto` for C++ and only use `__auto_type` for C. It shouldn't make a difference, but not relying on compiler specific extensions can help avoid weird edge cases and make things more standardized overall. You should be able to determine which to use by checking the `macroDefinitionRecord.Handle.Language`

Reply:
The problem with doing that check is that I should also check the C++ standard used (preC++11 code would not compile if we don't do that). I couldn't find any method in clang-c that gives me the language features of the cursor, and I also see that we don't save the used standard inside `PInvokeGeneratorConfiguration`.

Reply:
Please feel free to pass through the standard to the `PInvokeGeneratorConfiguration` if that helps determine the version to use.